### PR TITLE
Final changes

### PR DESCRIPTION
**Pull request by dbhynds**
## Summary of Changes

Make a few final changes to clean things up, make them consistent, and abstract a bit more. Also add code coverage reports. It adds a `Arrayable` and `JsonStringable` concerns to reduce code duplication.

- [x] Depends on https://github.com/packbackbooks/lti-1-3-php-library/pull/124

## Testing

- [x] I have added automated tests for my changes
- [x] I ran `composer test` before opening this PR
- [x] I ran `composer lint-fix` before opening this PR

### CODE REVIEW

File: src/DeepLinkResources/Icon.php [hunk 1]
```diff
@@ -18,6 +20,15 @@ public static function new(string $url, int $width, int $height): self
         return new Icon($url, $width, $height);
     }
 
+    public function getArray(): array
```
Comment (cophaso) [inline src/DeepLinkResources/Icon.php:23]: I'm guessing the `getArray()` is just how the model is returned? I was wondering why this was named so generically.

Comment (dbhynds) [reply src/DeepLinkResources/Icon.php:23]: Correct. I need something that builds the potential array of properties before sending in through the various concerns for processing. I settled on `getArray()` but am open to suggestions for a better way to name this.

**Review by cophaso — APPROVED** [review]
I think this looks good, I don't have access to the lti-library repo in code climate. I'm not sure if you just need to invite people. It looks like the upload ran successfully so I think it should be fine, I'm sure it would throw an error if the code coverage wasn't formatted like it expected. 

https://github.com/packbackbooks/lti-1-3-php-library/actions/runs/7268165657/job/19803484110

File: src/DeepLinkResources/Icon.php (hunk 1 above)
Comment (cophaso) [reply src/DeepLinkResources/Icon.php:23]: I think this is fine, I could understand what it was doing by just looking at the function.

**Pull request merged**